File: haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

divides :: Integer -> Integer -> Bool
divides n m = (n `mod` m) == 0

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum [x | x <- [1..limit-1],
               any (divides x) [f | f <- factors, f > 0]]

File: dart/sum-of-multiples/lib/sum_of_multiples.dart
class SumOfMultiples {

  int sum(List<int> factors, int limit) => Iterable<int>.generate(limit)
      .where((n) => factors.any((element) => element != 0 && n % element == 0))
      .fold(0, (acc, n) => acc + n);
}
